pkg/appmanager/as3Manager.go

```diff
@@ -369,18 +373,24 @@ func (appMgr *Manager) buildAS3Declaration(obj as3Object, template as3Template)
 // Takes AS3 Declaration and post it to BigIP
 func (appMgr *Manager) postAS3Declaration(declaration as3Declaration) {
 	log.Debugf("[as3_log] Processing AS3 POST call with AS3 Manager")
-	var as3RC As3RestClient
 	as3RC.baseURL = BigIPURL
-	response, _ := as3RC.restCallToBigIP("POST", "/mgmt/shared/appsvcs/declare", declaration, appMgr.sslInsecure)
-	log.Debugf("[as3_log] AS3 declaration POST call response %s", response)
+	as3RC.restCallToBigIP("POST", "/mgmt/shared/appsvcs/declare", declaration, appMgr.sslInsecure)
 
 }
 
 // Takes AS3 Declaration, method, API route and post it to BigIP
 func (as3RestClient *As3RestClient) restCallToBigIP(method string, route string, declaration as3Declaration, sslInsecure bool) (string, bool) {
 	log.Debugf("[as3_log] REST call with AS3 Manager")
-	timeout := time.Duration(15 * time.Second)
+	hash := md5.New()
+	io.WriteString(hash, string(declaration))
+	as3RestClient.newChecksum = string(hash.Sum(nil))
+	timeout := time.Duration(60 * time.Second)
 	var body []byte
+	if as3RestClient.oldChecksum == as3RestClient.newChecksum {
+		log.Debugf("[as3_log] No change in declaration.")
+		return string(body), true
+	}
+	as3RestClient.oldChecksum = as3RestClient.newChecksum
```

Comment:
This needs to be done after we send the declaration to BIG-IP and it is successful.